app/Http/Controllers/TicketController.php

```diff
@@ -36,13 +36,19 @@ public function show($id)
         //We need to ensure that this ticket belongs to this user
         $tickets = $this->getTickets();
         foreach ($tickets as $ticket) {
-            if ($ticket->getTicketID() == $id) {
-                $accountTicketController = new AccountTicketController();
-                $replies = array_reverse($accountTicketController->getReplies($ticket, 1));
-                //Clear the cache here, because you may see a ticket with ISP responses but the list may not show it yet
+            try {
+                if ($ticket->getTicketID() == $id) {
+                    $accountTicketController = new AccountTicketController();
+                    $replies = array_reverse($accountTicketController->getReplies($ticket, 1));
+                    //Clear the cache here, because you may see a ticket with ISP responses but the list may not show it yet
+                    $this->clearTicketCache();
+                    return view("pages.tickets.show", compact('replies', 'ticket'));
+                }
+            }catch (Exception $e) {
```

Comment:
1. Try to be more specific with the type of exception being caught. A generic catch on the `Exception` type would inadvertently swallow errors that we don't necessarily want to.
2. I'm not sure if clearing the cache and redirecting them to the index is better than say, clearing the cache and retrying to fetch the ticket once or something.
3. There should be a space before `catch`

Reply:
There's no point refetching the ticket if it's not found, it'll just fail again. We should show an error that the ticket couldn't be found and clear the cache though.

Reply:
I've made the recommended changes.